src/components/App.tsx

```diff
@@ -3,31 +3,54 @@ import Header from './Header';
 import MainContent from './MainContent';
 import Footer from './Footer';
 import ErrorBoundary from './ErrorBoundary';
-import { ModalWindowType, SearchBy } from './../model';
+import { useParams, useSearchParams, useNavigate } from 'react-router-dom';
+import { ModalWindowType, MovieData, SearchBy } from './../model';
 import ModalWindowOpener from './ModalWindowOpener';
 import MovieDetails from './MovieDetils';
-import { useGetMoviesQuery } from '../query/movies';
+import { useGetMoviesQuery, useGetMovieInfoQuery } from '../query/movies';
 import { useAppSelector } from '../store/hooks';
 import { selectSelectedMovie } from '../store/moviesSlice';
-import { SortOrder } from '../model/movies-query-params';
+import { Genre, SortOrder } from './../model';
 
 const App = () => {
   const selectedMovie = useAppSelector(selectSelectedMovie);
+  const { movieId } = useParams();
+  const navigate = useNavigate();
 
-  const [searchText, setSearchText] = useState('');
-  const [sortBy, setSortBy] = useState('');
   const [modalWindowType, setModalWindowType] = useState(ModalWindowType.None);
-  const [isViewMode, setIsViewMode] = useState(false);
-  const [selectedGenre, setSelectedGenre] = useState();
+
+  const [searchParams, setSearchParams] = useSearchParams('');
+  const search = searchParams.get('search');
+  const selectedGenre = searchParams.get('filter') as Genre;
+  const sortBy = searchParams.get('sortBy');
 
   const queryParams = {
-    search: searchText,
-    searchBy: SearchBy.Title,
-    sortOrder: SortOrder.Increase,
+    search: search,
     sortBy: sortBy,
     filter: selectedGenre,
   } as const;
-  const { data, error, isLoading, refetch } = useGetMoviesQuery(queryParams);
+
+  const getQueryObject = () => {
+    let result = {} as any;
+    for (let key in queryParams) {
+      if ((queryParams as any)[key]) {
+        result[key] = (queryParams as any)[key];
+      }
+    }
```

Comment:
You can write like this:
```js
const result = Object.entries(queryParams)
  .reduce(
    ([key, value], result) => value ? ({ ...result, [key]: value }) : result
  )
```
This is more reactive and thus more maintainable, but eventually it's matter of taste